src/Workspaces/Core/Portable/FindSymbols/FindReferences/Finders/NamedTypeSymbolReferenceFinder.cs

```diff
@@ -89,57 +89,7 @@ protected override bool CanFind(INamedTypeSymbol symbol)
             var namedTypeReferences = await FindReferencesInDocumentWorker(
                 namedType, document, semanticModel, cancellationToken).ConfigureAwait(false);
 
-            // Mark any references that are also Constructor references.  Some callers
-            // will want to know about these so they won't display duplicates.
-            return await MarkConstructorReferences(
-                namedType, document, semanticModel, namedTypeReferences, cancellationToken).ConfigureAwait(false);
-        }
-
-        private async Task<ImmutableArray<FinderLocation>> MarkConstructorReferences(
-            INamedTypeSymbol namedType, Document document,
-            SemanticModel semanticModel,
-            ImmutableArray<FinderLocation> namedTypeReferences,
-            CancellationToken cancellationToken)
-        {
-            var constructorReferences = ArrayBuilder<FinderLocation>.GetInstance();
-            foreach (var constructor in namedType.Constructors)
-            {
-                var references = await ConstructorSymbolReferenceFinder.Instance.FindAllReferencesInDocumentAsync(
-                    constructor, document, semanticModel, cancellationToken).ConfigureAwait(false);
-                constructorReferences.AddRange(references);
-            }
-
-            var result = ArrayBuilder<FinderLocation>.GetInstance();
-            foreach (var finderLocation in namedTypeReferences)
-            {
-                if (Contains(constructorReferences, finderLocation))
-                {
-                    var location = finderLocation.Location;
-                    location.IsDuplicateReferenceLocation = true;
```

Comment:
📝 This is a dead store to a field of a value type in a temporary location. It is immediately discarded.

Reply:
this looks like this regressed.  the code used to be:

```c#
                    var localReference = reference;
                    localReference.Location.IsDuplicateReferenceLocation = true;
                    result.Add(localReference);
```

This data would not have been lost.

Reply:
Looks like i regressed this in: https://github.com/dotnet/roslyn/commit/3993e8dd5d72c898e2008add8afd7226500b8d57

However, i would prefer we maintain the original semantics rather than just remove this code.  The logic here was intentional and was supposed to be something observable in the UI.

Reply:
@CyrusNajmabadi I'm happy to restore the original behavior as long as someone can explain what the user-facing bug is 😄

Reply:
@sharwell It may be reproable through the UI, or through the API (or both).  The original code came in through:

https://github.com/dotnet/roslyn/pull/14047

I think the issue is that if you search for a NamedType, you can get duplicated results for both it and its' constructors on an "new Foo()" expr.  

So, for example, if you have:

```c#
class Foo
{
    public Foo(int i)
    {

    }
}

class Bar
{
    void X()
    {
        new Foo(1);
    }
}
```

We show you:

![image](https://user-images.githubusercontent.com/4564579/55291866-b2b74380-5398-11e9-8c4d-ce7a24b65fb9.png)

But we decided we don't want that, and we intended for the "new" call to only show up in the constructors list.

--

yes, this is a fail on me to not adequately test.

Reply:
➡️ On further review, I believe the original internal API for this feature is lacking. Rather than restore it to what I believe is a suboptimal representation, I filed #35139 to follow-up on this by creating a clearer Find All References API that more clearly supports grouping locations by the referenced symbol, even when it differs from the original symbol searched for.